yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -271,6 +271,33 @@ private[spark] class Client(
         appContext.setResource(capability)
     }
 
+    sparkConf.get(ROLLED_LOG_INCLUDE_PATTERN).foreach { includePattern =>
```

Comment:
@jerryshao  on this old commit -- am I right that this outer foreach is not supposed to include all of the rest of the body here? it looks like for each include pattern, it will process all exclude patterns, and set the context again. are these supposed to be two separate loops or did I miss the point?

No worries if so, I may be able to change this if needed while making changes for 2.6.